presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -1684,10 +1585,15 @@ public void transferInitiated(Context.TransferInitiated context)
                 public void bytesTransferred(Context.BytesTransferred context)
                 {
                     long currentBytes = context.progressSnapshot().transferredBytes();
-                    if (currentBytes - previousBytesTransferred >= 10 * 1024 * 1024) {
-                        log.debug("Upload progress (%s/%s): %d bytes transferred",
-                                host, key, currentBytes);
+                    double ratioTransferred = context.progressSnapshot().ratioTransferred().orElse(0.0);
+
+                    // Logging every 10% progress or every 10MB
+                    if (ratioTransferred - previousRatio >= 0.1 ||
+                            currentBytes - previousBytesTransferred >= 10 * 1024 * 1024) {
```

Comment:
nit: I think we should only print every 10% progress. Otherwise, it will create unnecessary logs. In the log, we can print both the percentage completed and the number of bytes transferred.